pkgs/development/python-modules/devito/default.nix

```diff
@@ -0,0 +1,103 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, anytree
+, nbval
+, sympy
+, scipy
+, cached-property
+, psutil
+, py-cpuinfo
+, cgen
+, click
+, multidict
+, distributed
+, pyrevolve
+, codepy
+, pytestCheckHook
+, matplotlib
+, fetchpatch
+}:
+
+buildPythonPackage rec {
+  pname = "devito";
+  version = "4.6.2";
+
+  src = fetchFromGitHub {
+    owner = "devitocodes";
+    repo = "devito";
+    rev = "v${version}";
+    sha256 = "sha256-1yPKaVtxjeoANCUAqJYZndi9C/1BVVg5vuNNAb19zvs=";
+  };
+
+  # Small fix to avoid runtime errors
+  patches = [
+    # https://github.com/devitocodes/devito/pull/1850
+    (fetchpatch {
+      url = "https://github.com/devitocodes/devito/commit/b46a51dfdc67e511dcd2fb3eecc4cb8f5aa9afc2.patch";
+      hash = "sha256-Jhn0FrXD504LhHAUkYhphfqMwMiAk8EmXaZRGqT0vPA=";
+    })
+  ];
+
+  postPatch = ''
+    # Removing unecessary dependencies
+    substituteInPlace requirements.txt \
+    --replace "codecov" "" \
+    --replace "distributed<=2022.2" "distributed" \
+    --replace "flake8>=2.1.0" "" \
+    --replace "pytest>=3.6,<7.0" "" \
+    --replace "pytest-runner" "" \
+    --replace "pytest-cov" "" \
+    --replace "sympy>=1.7,<=1.9" "sympy" \
+  '';
+
+  checkInputs = [ pytestCheckHook matplotlib ];
+
+  disabledTests = [
+    "test_assign_parallel"
+    "test_gs_parallel"
+    "test_if_parallel"
+    "test_if_halo_mpi"
+    "test_cache_blocking_structure_distributed"
+    "test_mpi"
+    "test_codegen_quality0"
+    "test_new_distributor"
+    "test_subdomainset_mpi"
+    "test_init_omp_env_w_mpi"
+    "test_mpi_nocomms"
+  ];
+
+  disabledTestPaths = [
+    "tests/test_pickle.py"
+    "tests/test_benchmark.py"
+    "tests/test_mpi.py"
+    "tests/test_autotuner.py"
+    "tests/test_data.py"
+    "tests/test_dse.py"
+    "tests/test_gradient.py"
```

Comment:
Please add comments why the tests are disabled.

Reply:
It would be kind of a daunting task to grab every last error message but I guess I could add a comment saying that the tests are failing because of MPI issues, but they pass outside the nix-build.

Reply:
yeah, that would already be better than nothing to have some indication whats going on.